dookug-common/dookug-common-pdfsignature/src/main/java/hu/icellmobilsoft/common/openpdfsign/KeystoreLoader.java

```diff
@@ -0,0 +1,253 @@
+/*-
+ * #%L
+ * DookuG
+ * %%
+ * Copyright (C) 2023 - 2024 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.common.openpdfsign;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.enterprise.inject.Model;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JceOpenSSLPKCS8DecryptorProviderBuilder;
+import org.bouncycastle.pkcs.PKCS8EncryptedPrivateKeyInfo;
+
+import hu.icellmobilsoft.coffee.dto.exception.TechnicalException;
+import hu.icellmobilsoft.coffee.dto.exception.enums.CoffeeFaultType;
+import hu.icellmobilsoft.coffee.se.api.exception.BaseException;
+import hu.icellmobilsoft.coffee.se.api.exception.BusinessException;
+import hu.icellmobilsoft.dookug.common.util.random.RandomUtil;
+
+/**
+ * Keystore helper for creating PKCS12 keystore from PKCS8 files
+ * 
+ * @author tamas.cserhati
+ * @since 1.1.0
+ */
+@Model
+public class KeystoreLoader {
+
+    private static final String X_509 = "X.509";
+    private static final String PKCS12 = "PKCS12";
+    /**
+     * default key alias
+     */
+    public static final String DEFAULT_ALIAS = "alias";
+
+    /**
+     * this password is used for encrypting the temporaly generated PKCS12 keystore
+     */
+    private final char[] currentKeystorePassword = (RandomUtil.generateId() + RandomUtil.generateToken()).toCharArray();
```

Comment:
Miért kell ez a 2 random bele? A token nem elég? abban is van generateId hívva.

Reply:
igazad van valoban
csak egy hosszu  random stringet akartam generalni